src/main/webapp/assets/js/signUpScript.js

```diff
@@ -52,23 +69,195 @@ async function onLoad() {
+  emailForm = document.getElementById('email-form');
+  submitFormButton = document.getElementById('form-submit');
+  subscribeFormSubmitted = document.getElementById('subscribe-form-submitted');
+  unsubscribeFormSubmitted = document.getElementById('unsubscribe-form-submitted');
+  submitFormButton.addEventListener('click', submitForm);
+}
+
+/**
+ * On click of the submit button, validate form data and send data to the servlet.
+ * @return {void}
+ */
+async function submitForm() {
+  // Disable multiple submissions.
+  submitFormButton.disabled = true;
+
+  if (validateForm()) {
+    emailForm.action = '/user';
+
+    // Based on user subscribing or unsubscribing, show confirmation that form
+    // was submitted.
+    if (document.getElementById('subscribe').checked) {
+      subscribeFormSubmitted.style.display = 'block';
+    } else {
+      unsubscribeFormSubmitted.style.display = 'block';
+    }
+
+    // Waits for 4000 ms so user can read confirmation and submits the form.
+    setTimeout(function() {
+      emailForm.submit();
+    }, 4000);
+  } else {
+    submitFormButton.disabled = false;
+  }
+}
+
+/**
+ * Goes through the form elements and marks the invalid inputs.
+ * Returns whether all the inputs are valid.
+ * @return {boolean}
+ */
+function validateForm() {
+  const invalidIds = [];
+  const errorMessages = [];
+  const formElements = emailForm.elements;
+
+  disableInjection(formElements);
+  validateFormText(invalidIds, errorMessages, formElements);
+  validateFormEmail(invalidIds, errorMessages, formElements);
+  validateFormCollege(invalidIds, errorMessages, formElements);
+  validateFormRadioButtons(invalidIds, errorMessages, formElements);
+  markInvalidInputs(invalidIds, errorMessages, formElements);
+
+  return (invalidIds.length === 0);
+}
+
+/**
+ * Checks if the email is valid.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateFormEmail(invalidIds, errorMessages, formElements) {
+  const email = formElements.namedItem('email').value;
+  const emailPattern = /^[a-zA-Z0-9._-]+@google.com$/;
+  if (!emailPattern.test(email)) {
+    invalidIds.push('email');
+    errorMessages.push('email must follow @google.com');
+  }
+}
+
+/**
+ * Checks if at least one radio button has been checked.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateFormRadioButtons(invalidIds, errorMessages, formElements) {
+  const subscribe = formElements.namedItem('email-notif')[0].checked;
+  const unsubscribe = formElements.namedItem('email-notif')[1].checked;
+
+  if (!subscribe && !unsubscribe ) {
+    invalidIds.push('subscribe');
+    errorMessages.push('must pick to subscribe or unsubscribe');
+  }
+}
+
+/**
+ * Check if form text elements have been filled out completely.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
```

Comment:
for this, you could do
```js
/*
...
* @param {Array<string>} invalidIds - some description
* @param {Array<string>} errorMessages - some description
* @param {Array<HTMLElement>} formElements - some desription
*/
```
to be more specific! Also, having `@return {void}` is optional, since `void` doesn't really exist as a type in Javascript anyway (it's an operator that always returns `undefined` after evaluating a statement https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void)

Reply:
Gotcha! Will implement!